### PR TITLE
CON-4675

### PR DESCRIPTION
**Pull request by SimonBaeumer**
@sbofirov I fixed the problem with the cron jobs via the command line. Further I moved the subscriber registration into its own class but I am not completely sure how to test the doctrine events.

I test everything atm and it looks very good to me. It makes the bootstrap file ~300 lines smaller .:)

### CODE REVIEW

**Comment by sbofirov** [comment]
To test doctrine events, open some exported product and change his title. then save it and check sw_connect_change table. If product doesn't have variants connect plugin must generate only 1 change. If this products has variants connect plugin must generate change for every variant.

**Comment by sbofirov** [comment]
@SimonBaeumer will be so hard to test every single subscriber, but with checkout Connect product we can reach as much as possible.

* Export
* Import
* Create order with Connect product
* Order exists in both shops. It has Connect logo

**Comment by sbofirov** [comment]
I forget to ask, this changes won't be part of 1.0.18, right? We don't have enough time to test it.

File: Bootstrap/SubscriberRegistration.php [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+    }
+
+    /**
+     * @return Lifecycle
+     */
+    private function getLifecycleSubscriber()
+    {
+        if (!$this->lifecycle) {
+            $this->lifecycle = new Lifecycle(
+                Shopware()->Models(),
```
Comment (sbofirov) [inline Bootstrap/SubscriberRegistration.php]: Please replace that with ```$this->modelManager```

File: Bootstrap/SubscriberRegistration.php [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+        $subscribers = [
+            new \ShopwarePlugins\Connect\Subscribers\TemplateExtension(),
+            $this->createCheckoutSubscriber(),
+            new \ShopwarePlugins\Connect\Subscribers\Voucher(),
+            new \ShopwarePlugins\Connect\Subscribers\BasketWidget(),
+            new \ShopwarePlugins\Connect\Subscribers\Dispatches(),
+            new \ShopwarePlugins\Connect\Subscribers\Javascript(),
+            new \ShopwarePlugins\Connect\Subscribers\Less(),
+            $this->getLifecycleSubscriber()
+
```
Comment (sbofirov) [inline Bootstrap/SubscriberRegistration.php]: remove this empty line

File: Bootstrap/SubscriberRegistration.php [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+                    new ErrorHandler(),
+                    $this->eventManager
+                )
+            ),
+            new \ShopwarePlugins\Connect\Subscribers\Category(
+                $this->modelManager
+            ),
+            new \ShopwarePlugins\Connect\Subscribers\Connect(),
+            new \ShopwarePlugins\Connect\Subscribers\Payment(),
+            new \ShopwarePlugins\Connect\Subscribers\ServiceContainer(
```
Comment (SimonBaeumer) [inline Bootstrap/SubscriberRegistration.php]: Done

**Comment by SimonBaeumer** [comment]
@sbofirov It will be released in the 1.1.0 version

File: Bootstrap/SubscriberRegistration.php (hunk 2 above)
Comment (SimonBaeumer) [inline Bootstrap/SubscriberRegistration.php]: done

File: Bootstrap/SubscriberRegistration.php (hunk 1 above)
Comment (SimonBaeumer) [inline Bootstrap/SubscriberRegistration.php]: done

File: Bootstrap/SubscriberRegistration.php (hunk 3 above)
Comment (SimonBaeumer) [inline Bootstrap/SubscriberRegistration.php]: Done

**Pull request merged**